src/app.module.ts

```diff
@@ -0,0 +1,31 @@
+import { MongooseModule } from '@nestjs/mongoose';
+import { ConfigModule } from '@nestjs/config';
+
+import { UserModule } from './user/user.module';
+import {
+  MiddlewareConsumer,
+  Module,
+  NestModule,
+  RequestMethod,
+} from '@nestjs/common';
+import { AppController } from './app.controller';
+import { AppService } from './app.service';
+import { AuthenticationMiddleware } from './middlwares/authentication.middleware';
+
+@Module({
+  imports: [
+    ConfigModule.forRoot(),
+    UserModule,
+    MongooseModule.forRoot(process.env.MONGODB_URL),
+  ],
+  controllers: [AppController],
+  providers: [AppService],
+})
+export class AppModule implements NestModule {
+  configure(consumer: MiddlewareConsumer) {
```

Comment:
For Authentication this middleware is not needed. please refer to the comment in middleware.